### ISSUE TITLE
Bug: Incorrect numbers in transactions import summary

### ISSUE THREAD

**Issue by nicogaldamez** [maybe-finance/maybe#1317]
**Describe the bug**
At the final step of a transactions import, the summary showing how many new items will be added displays numbers that don’t match the options selected in the previous steps. 

I imported a CSV with 2 rows, mapped every category, tag, and account, but the summary shows that some of those items will be created.

If I go back to the mapping step, I can see that some of the selected options were cleared.

**To Reproduce**
1. Go to "Import"
2. Click on "Import transactions"
3. Paste rows separated by commas or upload a CSV
4. Click "Upload CSV"
5. Configure your import and click "Apply configuration"
6. Click "Next step" on the clean step
7. Select an existing category for each category in the list, then click "Next"
8. Select an existing tag for each tag in the list, then click "Next"
9. Select an existing account for each account in the list, then click "Next"

**Expected behavior**
I expected a zero next to Accounts, Categories, and Tags, but I see a number greater than 0.

**Screenshots / Recordings**
<img width="813" alt="image" src="https://github.com/user-attachments/assets/a812d920-e62b-44ae-bec1-c493648019c3">
<img width="795" alt="image" src="https://github.com/user-attachments/assets/4d059a18-3d65-4de8-b522-887e4fa2e76c">
<img width="805" alt="image" src="https://github.com/user-attachments/assets/1575e659-dc04-43aa-8c88-824795f46ca8">
<img width="695" alt="image" src="https://github.com/user-attachments/assets/8f89e59e-2842-4bbf-a97a-99bf8203ef5a">

If I go back to Map step I see that some options have been cleared:
<img width="716" alt="image" src="https://github.com/user-attachments/assets/09d6e784-d28f-4e40-968b-8485d07f0591">
<img width="707" alt="image" src="https://github.com/user-attachments/assets/2062dd45-55e5-40af-a75c-21ecc2ce1ce0">

**Comment by zachgoll** [comment]
@nicogaldamez good catch!  Will take a look.

**Comment by nicogaldamez** [comment]
To be honest, I wasn't sure if it was a bug or if I was just misunderstanding the expected behavior ☺️. But now that you've confirmed it's a bug, I'd love to help find and fix the issue.

Let me know if you'd like me to take a look, @zachgoll.

**Comment by zachgoll** [comment]
@nicogaldamez yeah, absolutely!  Feel free to take a look / open a PR, would be super helpful!

**Comment by nicogaldamez** [comment]
### TL;DR;
I found that the problem is related to the timeout in the `auto-submit-form` stimulus controller. It waits 500ms to submit the form, and if you click the 'Next' button quickly, the form isn't submitted, and the option isn't saved.

### Details
Each row in the mapping step is a form that auto-submits after an option is selected from the dropdown.

<img width="995" alt="image" src="https://github.com/user-attachments/assets/ce5f8d3d-dc03-4c47-9a8e-4db22e818421">

It uses auto-submit-form-controller to submit the form after 500ms:

```js
// javascript/controllers/auto_submit_form_controller.js
  handleInput = () => {
    clearTimeout(this.timeout);
    this.timeout = setTimeout(() => {
      this.element.requestSubmit();
    }, 500);
  };
```

If I select a category and click 'Next' before the form submits, the category doesn't get assigned.

### Potential solution

We could add a value to the stimulus controller to specify the wait time. In this case, we can set it to 0 to submit the form immediately.

By default, the wait time to submit the form could be 500ms.

An alternative is to set the wait time for each element in the form using a data attribute, and change the `handleInput` method to use that attribute. This way, we could have more flexibility.

Does this sound right to you? Which option do you think is better?

**Comment by zachgoll** [comment]
@nicogaldamez ahh that makes sense!

Yeah, I think I like a solution where we set a default debounce timeout based on the _type_ of the input.

For example, a `text_field` would default to `500ms` while a `select` would not be debounced at all.

It also appears that I may have incorrectly set the input event on this: `"data-autosubmit-trigger-event": "change"`

From my understanding, the `change` event only fires when the input loses focus.  I think we may want the `input` event for these so it fires immediately upon update.

That _should_ allow us to remove this arbitrary `sleep 1` call in the system test as well!

"data-autosubmit-trigger-event": "change"

**Comment by nicogaldamez** [comment]
@zachgoll thanks for the quick response!

>From my understanding, the change event only fires when the input loses focus. I think we may want the input event for these so it fires immediately upon update.

It's a select tag and the `change` event is fired when an option is selected, so we are good with that 👌

>Yeah, I think I like a solution where we set a default debounce timeout based on the type of the input.

Do you think adding logic to the stimulus controller to determine the type of the element is a better approach than using a data-attribute? I thought about using a data-attribute because it gives you more flexibility.

I was thinking something like this:
```ruby
 "data-auto-submit-form-target": "auto", 
 "data-autosubmit-trigger-event": "change", 
 "data-autosubmit-debounce-timeout": 0
```
